tests/tools/onert_train/src/nnfw_util.h

```diff
@@ -36,6 +36,9 @@ namespace onert_train
 uint64_t num_elems(const nnfw_tensorinfo *ti);
 uint64_t bufsize_for(const nnfw_tensorinfo *ti);
 
+std::string toString(const NNFW_TRAIN_OPTIMIZER &opt);
+std::string toString(const NNFW_TRAIN_LOSS &loss);
+std::string toString(const NNFW_TRAIN_LOSS_REDUCTION &loss_rdt);
```

Comment:
I have a plan to use this function later, here : 

 https://github.com/Samsung/ONE/pull/12445#r1466143798

Reply:
Oh, I didn't think much about this part. 

Either way(passing by copy, or cost reference) looks good to me. 

-------------------------------------------------------

IHMO, 
It is trivial but one good point of using `const reference` is that we can give consistency to the `toString` function. 

If we want to add `toString` function of another class/struct(not enum), later. 
We might make a function like `string toString(const type&)` to avoid an additional copy.
I think it is a kind of consistency - that a similar purpose function has a similar definition.

Reply:
> enum is okay to be copied

means that it's like `toString(int32_t loss_rdt);`. It's just copying 4 bytes memory. That's why I said **okay**. 😄

It depends on you's idea. For me, I cannot feel satisfied with the following code.
```cpp
// for consistency of function signature
// Is the following also okay?
std::string toString(const bool &sth_bool);
```

Reply:
Of course, it's a trivial.

Reply:
> // for consistency of function signature
> // Is the following also okay? 
> std::string toString(const bool &sth_bool);  

Hmm.. Definitely, it looks awkward. 
Your point makes sense :smile: 

I'd better update to remove the reference here.